core/domain/prod_validation_jobs_one_off.py

```diff
@@ -110,116 +288,3206 @@ def validate(cls, item):
+            item.contributors_summary.keys())
+        if sorted(item.contributor_ids) != sorted(
+                contributor_ids_from_contributors_summary):
+            cls.errors['contributors summary check'] = (
+                'Model id %s: Contributor ids: %s do not match the contributor '
+                'ids obtained using contributors summary: %s') % (
+                    item.id, (',').join(sorted(item.contributor_ids)),
+                    (',').join(
+                        sorted(contributor_ids_from_contributors_summary)))
+
+    @classmethod
+    def _validate_language_code(cls, item):
+        """Validate that language code is present in allowed language codes.
+
+        Args:
+            item: ExpSummaryModel to validate.
+        """
+        allowed_language_codes = [
+            language_item['code'] for language_item in (
+                constants.ALL_LANGUAGE_CODES)]
+
+        if item.language_code not in allowed_language_codes:
+            cls.errors['language code check'] = (
+                'Model id %s: Language code %s for model is unsupported' % (
+                    item.id, item.language_code))
+
+    @classmethod
+    def _validate_first_published_msec(cls, item):
+        """Validate that first published time of model is less than current
+        time.
+
+        Args:
+            item: ExpSummaryModel to validate.
+        """
+        if not item.first_published_msec:
+            return
+
+        epoch = datetime.datetime.utcfromtimestamp(0)
+        current_msec = (
+            datetime.datetime.utcnow() - epoch).total_seconds() * 1000.0
+        if item.first_published_msec > current_msec:
+            cls.errors['first published msec check'] = (
+                'Model id %s: The first_published_msec field has a value %s '
+                'which is greater than the time when the job was run'
+                ) % (item.id, item.first_published_msec)
+
+    @classmethod
+    def _validate_related_model_properties(cls, item):
+        """Validate that model properties match the corresponding exploration
+        model and exploration rights model properties.
+
+        Args:
+            item: ExpSummaryModel to validate.
+        """
+        _, exploration_model_tuples = cls.external_models[
+            'exploration_model']
+        _, exploration_rights_model_tuples = cls.external_models[
+            'exploration_rights_model']
+        exploration_model = exploration_model_tuples[0][1]
+        exploration_rights_model = exploration_rights_model_tuples[0][1]
+
+        if item.title != exploration_model.title:
+            cls.errors['title field check'] = (
+                'Model id %s: title field in model: %s does not match '
+                'corresponding exploration title field: %s') % (
+                    item.id, item.title, exploration_model.title)
+
+        if item.category != exploration_model.category:
+            cls.errors['category field check'] = 
```

Comment:
Add more information in this error: lengths of columns and topics list.

Reply:
Done.